### ISSUE TITLE
E-Mail restore doesn't work

### ISSUE THREAD

**Issue by OriginalXiaomi** [mailcow/mailcow-dockerized#2593]
Greetings!

First up: Mailcow is awesome. Thanks a lot for making my life so much easier.

When I switched my server to another hoster, everything worked out, but now every SOGo account doesn't have the hostname, it looks like that:
![image](https://user-images.githubusercontent.com/47394624/57559818-021f6500-7384-11e9-852f-3f7f2b28a996.png)

Also, I restored all my data, but no emails show up in SOGo or in gmail. I've double checked and they're actually in /var/lib/docker/volumes/mailcowdockerized_vmail-vol-1, but still, they aren't showing up in SOGo.

Thanks a lot in advance 👍 

Edit: I don't really care about the IP address in SOGo as is works with or without it. The emails don't show up anyways.

**Comment by OriginalXiaomi** [comment]
I can receive and write emails, so I think the problem is in linking the data from vmail to mailcow.

**Comment by andryyy** [comment]
Hi, you are welcome! :)

If you didn’t see an IP there before, you didn’t update mailcow before moving it. Newer mailcows list the IP instead of the hostname.
I don’t really think that’s the problem here anyway, just to clarify why you see the IP.

I cannot help any further, as you deleted the issue template and didn’t provide any container logs. :/

Please attach sogo-mailcow, dovecot-mailcow and mysql-mailcow logs. Limit the logs to the time you try to access your mails. Don’t post all logs, please. :)

How did you move your installation by the way?

**Comment by OriginalXiaomi** [comment]
I moved my installation as following:
1. Installed mailcow on new server
2. Created backup of everything using the helper-script
3. Restored everything using the helper-script

MariaDB wouldn't let me import the SQL-data using the helper-script, so I did it manually and it worked (except for emails).

sogo-mailcow:
`sogo-mailcow_1       | May 11 13:40:16 5130f4d29a50 sogod [72]: SOGoRootPage successful login from 'MYIPADDRESS' for user 'angelmann@grossartig.group' - expire = -1  grace = -1
sogo-mailcow_1       | May 11 13:40:16 5130f4d29a50 sogod [72]: MYIPADDRESS "POST /SOGo/connect HTTP/1.0" 200 53/89 0.008 - - 0
sogo-mailcow_1       | May 11 13:40:17 5130f4d29a50 sogod [72]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group HTTP/1.0" 302 0/0 0.003 - - 0
sogo-mailcow_1       | May 11 13:40:17 5130f4d29a50 sogod [72]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/view HTTP/1.0" 302 0/0 0.004 - - 0
sogo-mailcow_1       | May 11 13:40:17 5130f4d29a50 sogod [72]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail HTTP/1.0" 302 0/0 0.004 - - 0
sogo-mailcow_1       | May 11 13:40:17 5130f4d29a50 sogod [72]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail/view HTTP/1.0" 200 27122/0 0.058 139168 80% 2M
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [72]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Calendar/alarmslist?browserTime=1557574830 HTTP/1.0" 200 63/0 0.010 - - -1356K
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [72]: <0x0x556cfbfd6890[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [69]: <0x0x556cfc124620[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [73]: <0x0x556cfc035320[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [66]: [ERROR] <0x0x556cfbdaf550[NGImap4ConnectionManager]> IMAP4 login failed:
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 host=mail.grossartig.group, user=moin@grossartig.group, pwd=yes
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 url=imap://moin%40grossartig.group@mail.grossartig.group/
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 base=(null)
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 base-class=(null))
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 = <0x0x556cfbfe8e90[NGImap4Client]: login=moin@grossartig.group(pwd) socket=<NGActiveSocket[0x0x556cfbfe34e0]: mode=rw address=<0x0x556cfbfe3550[NGInternetSocketAddress]: host=5130f4d29a50 port=38774> connectedTo=<0x0x556cfbfe9020[NGInternetSocketAddress]: host=mail.grossartig.group port=143>>>
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [66]: <0x556cfbfdbf20[SOGoMailAccount]:5> renewing imap4 password
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [70]: <0x0x556cfc0426c0[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [66]: [ERROR] <0x0x556cfbdaf550[NGImap4ConnectionManager]> IMAP4 login failed:
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 host=mail.grossartig.group, user=moin@grossartig.group, pwd=yes
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 url=imap://moin%40grossartig.group@mail.grossartig.group/
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 base=(null)
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 base-class=(null))
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 = <0x0x556cfbe51650[NGImap4Client]: login=moin@grossartig.group(pwd) socket=<NGActiveSocket[0x0x556cfbfeccb0]: mode=rw address=<0x0x556cfbf0fc50[NGInternetSocketAddress]: host=5130f4d29a50 port=38778> connectedTo=<0x0x556cfbfeb2a0[NGInternetSocketAddress]: host=mail.grossartig.group port=143>>>
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [66]: [ERROR] <0x556cfbfdbf20[SOGoMailAccount]:5> Could not connect IMAP4
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [66]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail/5/view HTTP/1.0" 200 17/0 0.093 - - 0
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [68]: <0x0x556cfb7c71d0[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [69]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail/1/view HTTP/1.0" 200 932/0 0.350 - - 0
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [72]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail/0/view HTTP/1.0" 200 944/0 0.382 - - 0
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [73]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail/3/view HTTP/1.0" 200 932/0 0.414 - - 0
sogo-mailcow_1       | May 11 13:40:18 5130f4d29a50 sogod [72]: <0x0x556cfc19ae40[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:19 5130f4d29a50 sogod [72]: MYIPADDRESS "POST /SOGo/so/angelmann@grossartig.group/Mail/0/folderINBOX/view HTTP/1.0" 200 259/48 0.411 - - 0
sogo-mailcow_1       | May 11 13:40:19 5130f4d29a50 sogod [68]: MYIPADDRESS "GET /SOGo/so/angelmann@grossartig.group/Mail/2/view HTTP/1.0" 200 1387/0 0.825 - - 1M
sogo-mailcow_1       | May 11 13:40:19 5130f4d29a50 sogod [68]: <0x0x556cfbe00a70[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 sogod [70]: [ERROR] <0x0x556cfbdaf4e0[NGImap4ConnectionManager]> IMAP4 login failed:
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 host=mail.grossartig.group, user=marius@angelmann.net, pwd=yes
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 url=imaps://marius%40angelmann.net@mail.grossartig.group/?tls=YES
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 base=(null)
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 base-class=(null))
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 = <0x0x556cfc0426c0[NGImap4Client]: login=marius@angelmann.net(pwd) socket=<NGActiveSSLSocket[0x0x556cfc0cb670]: mode=rw address=(null)>>
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 sogod [70]: <0x556cfc079ea0[SOGoMailAccount]:4> renewing imap4 password
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 sogod [70]: <0x0x556cfc07e6e0[NGImap4Client]> TLS started successfully.
sogo-mailcow_1       | May 11 13:40:20 5130f4d29a50 sogod [68]: <0x0x556cfbe7d0c0[NGImap4Client]> TLS started successfully.`

Seems like SOGo isn't able to connect to IMAP.

dovecot-mailcow:
`dovecot-mailcow_1    | May 11 13:40:18 mail dovecot: imap(support@grossartig.group)<20628><mePEJZuIJJ2sFgEB>: Logged out in=154 out=1671 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:18 mail dovecot: imap(angelmann@grossartig.group)<20624><k6TEJZuIIrysFgH4>: Logged out in=154 out=1689 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:18 mail dovecot: imap(google@grossartig.group)<20632><4wzGJZuIPJ2sFgEB>: Logged out in=154 out=1679 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:19 mail dovecot: imap(angelmann@grossartig.group)<20637><eYrLJZuIaLysFgH4>: Logged out in=219 out=1736 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:20 mail dovecot: imap-login: Login: user=<gv.test@grossartig.group>, method=PLAIN, rip=5.230.23.75, lip=172.22.1.250, mpid=20645, TLS, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:20 mail dovecot: imap(gv.test@grossartig.group)<20645><POziJZuI0JEF5hdL>: Connection closed (EXPUNGE finished 0.005 secs ago) in=76 out=853 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:24 mail dovecot: imap-login: Login: user=<support@grossartig.group>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, mpid=20648, TLS, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:25 mail dovecot: imap-login: Login: user=<google@grossartig.group>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, mpid=20651, TLS, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:26 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 8 secs): user=<isabel@grossartig.group>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:26 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 8 secs): user=<marius@angelmann.net>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:26 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 2 secs): user=<isabel@grossartig.group>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:26 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 6 secs): user=<marius@angelmann.net>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 6 secs): user=<marius@angelmann.net>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 0 secs): user=<moin@grossartig.group>, rip=172.22.1.1, lip=172.22.1.250
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 8 secs): user=<marius@angelmann.net>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 0 secs): user=<moin@grossartig.group>, rip=172.22.1.1, lip=172.22.1.250
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap(google@grossartig.group)<20651><fNEqJpuIhJ2sFgEB>: Logged out in=145 out=1572 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap(angelmann@grossartig.group)<20640><XF/aJZuIcLysFgH4>: Logged out in=145 out=1564 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 4 secs): user=<isabel@grossartig.group>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 2 secs): user=<isabel@grossartig.group>, method=PLAIN, rip=172.22.1.1, lip=172.22.1.250, TLS: Connection closed, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:33 mail dovecot: imap(support@grossartig.group)<20648><tq/fJZuIcJ2sFgEB>: Logged out in=145 out=1556 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
dovecot-mailcow_1    | May 11 13:40:50 mail dovecot: imap-login: Login: user=<gv.test@grossartig.group>, method=PLAIN, rip=5.230.23.75, lip=172.22.1.250, mpid=20660, TLS, TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
dovecot-mailcow_1    | May 11 13:40:51 mail dovecot: imap(gv.test@grossartig.group)<20660><bdqyJ5uI9JEF5hdL>: Connection closed (EXPUNGE finished 0.005 secs ago) in=76 out=853 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0`

mysql-mailcow:
Did absolutely nothing since midnight. When I used it the last time.

`mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] mysqld (mysqld 10.2.24-MariaDB-1:10.2.24+maria~bionic) starting as process 1 ...
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Uses event mutexes
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Compressed tables use zlib 1.2.11
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Using Linux native AIO
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Number of pools: 1
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Using SSE2 crc32 instructions
mysql-mailcow_1      | 2019-05-11  0:43:39 139908822746432 [Note] InnoDB: Initializing buffer pool, total size = 256M, instances = 1, chunk size = 128M
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: Completed initialization of buffer pool
mysql-mailcow_1      | 2019-05-11  0:43:40 139908098565888 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: Highest supported file format is Barracuda.
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: 128 out of 128 rollback segments are active.
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: Creating shared tablespace for temporary tables
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: Waiting for purge to start
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] InnoDB: 5.7.26 started; log sequence number 3925141
mysql-mailcow_1      | 2019-05-11  0:43:40 139907863398144 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] Server socket created on IP: '::'.
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Warning] 'proxies_priv' entry '@% root@93d38b6c4642' ignored in --skip-name-resolve mode.
mysql-mailcow_1      | 2019-05-11  0:43:40 139908186572544 [Note] Event Scheduler: scheduler thread started with id 6
mysql-mailcow_1      | 2019-05-11  0:43:40 139908822746432 [Note] mysqld: ready for connections.
mysql-mailcow_1      | Version: '10.2.24-MariaDB-1:10.2.24+maria~bionic'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
mysql-mailcow_1      | 2019-05-11  0:43:41 139907863398144 [Note] InnoDB: Buffer pool(s) load completed at 190511  0:43:41`

When I tried to import SQL via the helper-script, it didn't quite work so I decided to restore it manually (described as in the documents) and it worked. Before I couldn't login and then I could, but the e-mails aren't there anyways. 

Thanks for your help :)

**Comment by andryyy** [comment]
Probably wrong MySQL credentials. Try to reset them as described in the docs. If you did it manually somehow, it probably broke or they changed. 

Make sure mailcow.conf is a symlink to .env too. 

I think it is just a SQL credentials issue. If not, I’m not able to help you here.

**Comment by OriginalXiaomi** [comment]
Okay, so.. Tried to restore SQL-data again and got this:

`mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)'
Check that mysqld is running and that the socket: '/var/run/mysqld/mysqld.sock' exists!
2019-05-11 17:03:57 139634009343296 [Note] mysqld (mysqld 10.2.24-MariaDB-1:10.2.24+maria~bionic) starting as process 6 ...
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Uses event mutexes
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Compressed tables use zlib 1.2.11
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Using Linux native AIO
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Number of pools: 1
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Using SSE2 crc32 instructions
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Initializing buffer pool, total size = 256M, instances = 1, chunk size = 128M
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Completed initialization of buffer pool
2019-05-11 17:03:57 139633280214784 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Highest supported file format is Barracuda.
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: 128 out of 128 rollback segments are active.
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Creating shared tablespace for temporary tables
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2019-05-11 17:03:57 139634009343296 [Note] InnoDB: 5.7.26 started; log sequence number 4004007
2019-05-11 17:03:57 139632851273472 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2019-05-11 17:03:57 139634009343296 [Note] Plugin 'FEEDBACK' is disabled.
2019-05-11 17:03:57 139634009343296 [Note] Server socket created on IP: '::'.
2019-05-11 17:03:57 139634009343296 [Note] Reading of all Master_info entries succeded
2019-05-11 17:03:57 139634009343296 [Note] Added new Master_info '' to hash table
2019-05-11 17:03:57 139634009343296 [Note] mysqld: ready for connections.
Version: '10.2.24-MariaDB-1:10.2.24+maria~bionic'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
2019-05-11 17:03:58 139632851273472 [Note] InnoDB: Buffer pool(s) load completed at 190511 17:03:58
mysqld is alive
Restoring...
/bin/sh: 1: cannot open backup/backup_mysql.gz: Permission denied
2019-05-11 17:04:00 139633821529856 [Note] mysqld (initiated by: [root] @ localhost []): Normal shutdown
2019-05-11 17:04:00 139633229858560 [Note] InnoDB: FTS optimize thread exiting.`

**Comment by nk-87** [comment]
For the database - I believe the username and password in the mailcow.conf file of your backup needs to match the credentials on the host's mailcow.conf. 

For the missing emails, if you're coming from an old or long running host, it restores them to ` /var/lib/docker/volumes/mailcowdockerized_vmail-vol-1/_data/<domain>/<mailbox>`, but it needs to go into another directory within it: `/var/lib/docker/volumes/mailcowdockerized_vmail-vol-1/_data/<domain>/<mailbox>/Maildir`.

I had to empty Maildir and move all the folder directories and Dovecot files in it to get my mail to show up.

**Comment by Braintelligence** [comment]
@OriginalXiaomi Did you solve your problem or find a workaround?

**Comment by bobskee** [comment]
> For the database - I believe the username and password in the mailcow.conf file of your backup needs to match the credentials on the host's mailcow.conf.
> 
> For the missing emails, if you're coming from an old or long running host, it restores them to ` /var/lib/docker/volumes/mailcowdockerized_vmail-vol-1/_data/<domain>/<mailbox>`, but it needs to go into another directory within it: `/var/lib/docker/volumes/mailcowdockerized_vmail-vol-1/_data/<domain>/<mailbox>/Maildir`.
> 
> I had to empty Maildir and move all the folder directories and Dovecot files in it to get my mail to show up.

This fixed it for me! Thanks

**Comment by andryyy** [comment]
yep, copy the mailcow.conf! :)

It will prepend or leave the Maildir prefix. It is all in your original mailcow.conf

**Comment by johanvandyck** [comment]
Hi I stumbled on the same issue. I was forced to reinstall my mailcow from scratch, but while importing I encountered the above bug/feature of the restore function.

_For the missing emails, if you're coming from an old or long running host, it restores them to /var/lib/docker/volumes/mailcowdockerized_vmail-vol-1/_data/<domain>/<mailbox>, but it needs to go into another directory within it: /var/lib/docker/volumes/mailcowdockerized_vmail-vol-1/_data/<domain>/<mailbox>/Maildir._

I like to make a resume, as the above text was not clear to me on first read.
- old installation has in mailcow.conf file the parameter "Maildir_sub="
- Newer installation has in mailcow.conf the parameter "Maildir_sub=Maildir"
- a feature / bug in the restore function doesn't take this parameter into account. It will restore in the same $Maildir_sub during the time the backup was taken. If this param is different now, the mail is restored, but not visible in the mailboxes.
- workaround: make sure the Maildir_sub is the same as during backup; restore; force resync during restore or afterwarts.
It would be nice if the restore function is adapted or at least warns the user for this issue.

Regards

Johan.
